### ISSUE TITLE
Segmentation fault on 64-bit *nix systems

### ISSUE THREAD

**Issue by rviscarra** [zeromq/fszmq#112]
This code fails on FreeBSD (64-bit mono) with a segfault:

```
#r "fszmq.dll"
open fszmq

let data  = Array.zeroCreate<byte> 10

let msg1 = new Message (data)

let data1 = Message.data msg1

printfn "My data: %A" data1
```

The bug is caused by `ZMQ_MSG_T_SIZE` being defined as 32 bytes. In newer versions of libzmq, message size is 64 bytes.

This issue also caused issues on other *nixes, specifically Arch Linux running 64-bit mono.

I submitted a pull request to address this issue: https://github.com/zeromq/fszmq/pull/111

**Comment by pblasucci** [comment]
Knew this was coming down the pipe. I was hoping to get to issue #110 before tackling support for 4.1.x of libzmq. There are a couple of breaking changes that have to be considered carefully. Specifically, both the change to zmq_msg_t size and the new way event monitoring messages are framed. Basically, I think we need to support 4.0.x and 4.1.x, which will mean doing some "smart" coding based on calls to `ZMQ.version`.

**Comment by pblasucci** [comment]
See also issue #113.

**Comment by pblasucci** [comment]
Addressed via #122.
